src/overlays/actors/ovl_Bg_Spot01_Idomizu/z_bg_spot01_idomizu.c

```diff
@@ -29,13 +28,66 @@ const ActorInit Bg_Spot01_Idomizu_InitVars = {
     (ActorFunc)BgSpot01Idomizu_Update,
     (ActorFunc)BgSpot01Idomizu_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idomizu/BgSpot01Idomizu_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idomizu/BgSpot01Idomizu_Destroy.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
+
+extern Gfx D_060007D0[];
+
+void BgSpot01Idomizu_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot01Idomizu* this = THIS;
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    if ((gSaveContext.eventChkInf[6] & 0x80) != 0 || LINK_AGE_IN_YEARS == YEARS_ADULT) {
+        this->waterHeight = -550.0f;
+    } else {
+        this->waterHeight = 52.0f;
+    }
+    this->actionFunc = func_808ABB84;
+    this->actor.posRot.pos.y = this->waterHeight;
+}
+
+void BgSpot01Idomizu_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void func_808ABB84(BgSpot01Idomizu* this, GlobalContext* globalCtx) {
+    if ((gSaveContext.eventChkInf[6] & 0x80) != 0) {
+        this->waterHeight = -550.0f;
+    }
+    globalCtx->colCtx.stat.colHeader->waterBoxes[0].unk_02 = this->actor.posRot.pos.y;
+    if (this->waterHeight < this->actor.posRot.pos.y) {
+        Audio_PlaySoundGeneral(NA_SE_EV_WATER_LEVEL_DOWN - SFX_FLAG, &D_801333D4, 4U, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+    }
+    Math_SmoothScaleMaxF(&this->actor.posRot.pos.y, this->waterHeight, 1.0f, 2.0f);
+}
+
+void BgSpot01Idomizu_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgSpot01Idomizu* this = THIS;
+    this->actionFunc(this, globalCtx);
+}
+
+void BgSpot01Idomizu_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    u32 framesTemp;
+    u32 yFramesComponent;
+    u32 xFramesComponent;
+    GraphicsContext* gfxCtx;
+    Gfx* dispRefs[2];
+
+    gfxCtx = globalCtx->state.gfxCtx;
+    Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_bg_spot01_idomizu.c", 228);
+    func_80093D84(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idomizu/func_808ABB84.s")
+    gSPMatrix(gfxCtx->polyXlu.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_spot01_idomizu.c", 232),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot01_Idomizu/BgSpot01Idomizu_Update.s")
+    framesTemp = globalCtx->state.frames;
+    yFramesComponent = framesTemp;
+    xFramesComponent = yFramesComponent % 128;
+    gSPSegment(gfxCtx->polyXlu.p++, 0x08,
+               Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 127 - xFramesComponent, yFramesComponent & 0x7F, 32, 32, 1,
```

Comment:
Can the `& 0x7F`s here become `% 128`?

Reply:
Unfortunately not. It doesn't match with `% 128`

Reply:
To be specific, the diff becomes this when you do `% 128` here

![image](https://user-images.githubusercontent.com/12245827/89479715-544a6400-d748-11ea-94e8-0363dc706643.png)

Reply:
Weird, okay. Well that may explain why I had such trouble matching this :P